pegen/pegen.c

```diff
@@ -857,6 +869,46 @@ name_default_pair(Parser *p, arg_ty arg, expr_ty value)
     return a;
 }
 
+/* Extracts all keys from an asdl_seq* of KeyValuePair*'s */
+asdl_seq *
+get_keys(Parser *p, asdl_seq *seq)
+{
+    int len = asdl_seq_LEN(seq);
+    asdl_seq *new_seq = _Py_asdl_seq_new(len, p->arena);
+    if (!new_seq) {
+        return NULL;
+    }
+    for (int i = 0; i < len; i++) {
+        KeyValuePair *pair = asdl_seq_GET(seq, i);
+        asdl_seq_SET(new_seq, i, pair->key);
+    }
+    return new_seq;
+}
+
+/* Extracts all values from an asdl_seq* of KeyValuePair*'s */
+asdl_seq *
+get_values(Parser *p, asdl_seq *seq)
+{
+    int len = asdl_seq_LEN(seq);
+    asdl_seq *new_seq = _Py_asdl_seq_new(len, p->arena);
+    if (!new_seq) {
+        return NULL;
+    }
+    for (int i = 0; i < len; i++) {
+        KeyValuePair *pair = asdl_seq_GET(seq, i);
+        asdl_seq_SET(new_seq, i, pair->value);
+    }
+    return new_seq;
+}
+
+int
+is_async(void *keyword)
```

Comment:
Nit (feel free to ignore): Should this function be public? If not, can you mark it as `static`? Otherwise, should it make sense to add it to the `doc`?

Reply:
Actually, with #111 we can remove this function altogether.